rest/config_test.go

```diff
@@ -1377,49 +1378,84 @@ func javaScriptHttpErrorEndpoint() func() (path string, teardownFn func()) {
 	}
 }
 
+// javaScriptHttpsEndpoint returns a callable function to setup an HTTPS endpoint that exposes
+// the given JavaScript source and returns a teardown function to terminate the underlying
+// httptest server instance.
+func javaScriptHttpsEndpoint(t *testing.T, js string) func() (path string, teardownFn func()) {
+	return func() (path string, teardownFn func()) {
+		ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, err := fmt.Fprint(w, js)
+			require.NoError(t, err)
+		}))
+		teardownFn = func() { ts.Close() }
+		return ts.URL, teardownFn
+	}
+}
+
 func TestLoadJavaScript(t *testing.T) {
 	const js = `function (doc, oldDoc) { if (doc.published) { channel("public"); } }`
 	var tests = []struct {
-		name        string
-		jsInput     func() (path string, teardownFn func())
-		jsExpected  string
-		errExpected error
+		name               string
+		jsInput            func() (path string, teardownFn func())
+		insecureSkipVerify bool
+		jsExpected         string
+		errExpected        error
 	}{
 		{
-			name:        "Load inline JavaScript",
-			jsInput:     inlineJavaScript(js),
-			jsExpected:  js,
-			errExpected: nil,
+			name:               "Load inline JavaScript",
+			jsInput:            inlineJavaScript(js),
+			insecureSkipVerify: false,
+			jsExpected:         js,
+			errExpected:        nil,
 		},
 		{
-			name:        "Load JavaScript from an external HTTP endpoint",
-			jsInput:     javaScriptHttpEndpoint(t, js),
-			jsExpected:  js,
-			errExpected: nil,
+			name:               "Load JavaScript from an external http endpoint",
+			jsInput:            javaScriptHttpEndpoint(t, js),
+			insecureSkipVerify: false,
+			jsExpected:         js,
+			errExpected:        nil,
 		},
 		{
-			name:        "Load JavaScript from an external endpoint that returns an error",
-			jsInput:     javaScriptHttpErrorEndpoint(),
-			jsExpected:  "",
-			errExpected: errInternalServerError,
+			name:               "Load JavaScript from an external https endpoint with ssl verification enabled",
+			jsInput:            javaScriptHttpsEndpoint(t, js),
+			insecureSkipVerify: false,
+			jsExpected:         "",
+			errExpected:        &url.Error{},
```

Comment:
If it's possible to have SG detect the specific error and convert it to a SG error type (that this test could check for), it would be preferable.

Reply:
Yeah, that's possible. Included changes to wrap the underlying certificate verification error in httpSSLVerifyError. This error type can be asserted in the unit test.